### PR TITLE
[GraphQL/Package Resolver] Limits for Move Type/Value Resolution

### PR DESCRIPTION
**Pull request by amnn**
## Description

Follow up from #15393, introducing four new limits to be imposed during Move type/value/layout resolution.  These are particularly relevant when handling the types that originate from user input, rather than on-chain values.

This PR simply introduces the limit configuration, but does not (yet) implement the limits.

## Test Plan

```
sui-graphql-rpc$ cargo nextest run
```

## Stack

- #15393 
- #15407 
- #15408
- #15409 
- #15410

### CODE REVIEW

File: crates/sui-graphql-rpc/src/config.rs [hunk 1]
```diff
@@ -16,6 +16,10 @@ const MAX_QUERY_PAYLOAD_SIZE: u32 = 2_000;
 const MAX_DB_QUERY_COST: u64 = 20_000; // Max DB query cost (normally f64) truncated
 const DEFAULT_PAGE_SIZE: u64 = 20; // Default number of elements allowed on a page of a connection
 const MAX_PAGE_SIZE: u64 = 50; // Maximum number of elements allowed on a page of a connection
+const MAX_TYPE_ARGUMENT_DEPTH: u32 = 16;
+const MAX_TYPE_ARGUMENT_WIDTH: u32 = 32;
```
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/config.rs:20]: Comment: was wondering where we got these default values from, looks like all but the max_type_argument_width have corresponding default values in sui-protocol-config. I feel like a generic type with 32 params would be crazy, so seems sensible enough to me lol

**Review by wlmyng — APPROVED** [review]
yeet

File: crates/sui-graphql-rpc/src/config.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-graphql-rpc/src/config.rs:20]: `max_type_argument_width` is also from our protocol configs or VM limits, under a different name:

https://github.com/MystenLabs/sui/blob/6db17d1d3b4565d8b91b3419d470adffda137eb7/crates/sui-protocol-config/src/lib.rs#L1192

Comment (wlmyng) [reply crates/sui-graphql-rpc/src/config.rs:20]: ah i see

**Comment by amnn** [comment]
This PR will be the next base to stage lands into `main`.

**Pull request merged**